fieldservice_maintenance/models/fsm_equipment.py

```diff
@@ -0,0 +1,51 @@
+# Copyright (C) 2018 Open Source Integrators
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class FSMEquipment(models.Model):
+    _inherit = "fsm.equipment"
+
+    maintenance_equipment_id = fields.Many2one(
+        "maintenance.equipment",
+        string="Related Maintenance Equipment",
+        readonly=True,
+        ondelete="restrict",
+        delegate=True,
+        auto_join=True,
+        index=True,
```

Comment:
You need required=True for delegate fields.
If the reason is that the field becomes required on the Form view, you can adapt the attrs required with False.